src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -67,9 +67,29 @@ public char Delimiter
         }
 
         /// <summary>
-        /// NoTypeInformation : should the #TYPE line be generated
+        /// IncludeTypeInformation : The #TYPE line should be generated. Default is false. Cannot specify with NoTypeInformation.
         /// </summary>
         [Parameter]
+        [Alias("ITI")]
+        public SwitchParameter IncludeTypeInformation
+        {
+            get
+            {
+                return _includeTypeInformation;
+            }
+            set
+            {
+                _includeTypeInformation = value;
+                _includeTypeInformationIsSet = true;
+            }
+        }
+        private bool _includeTypeInformation;
+        private bool _includeTypeInformationIsSet;
```

Comment:
In ProcessRecord() we use only `if (NoTypeInformation == false)`.
So I believe we should use auto-implemented property for `IncludeTypeInformation` and `NoTypeInformation` and check `IncludeTypeInformation.IsPresent`.

Reply:
I did testing with that.. and... it resulted in conflicts when using @splatting and supplying both. :( I ultimately went with this method because it was the most reliable way to detect if both were supplied.

Reply:
As for @splatting - maybe it is a bug in @splatting ?
I think we should remove ` NoTypeInformation` everywhere in code and leave only its declaration - the parameter is dummy and we should ignore it in code at all.

Reply:
It's not a bug in splatting. Its the nature of how switch params work. `IsPresent` is does not mean `was supplied` but more `user supplied $true`

```powershell
Function Test-SwitchParam {
    param(
        [Switch]$Switch
    )
    
    "Switch: {0}" -f $Switch
    "Switch.IsPresent: {0}" -f $Switch.IsPresent
    "  "
}

'Test-SwitchParam'
Test-SwitchParam
'Test-SwitchParam -Switch'
Test-SwitchParam -Switch
'Test-SwitchParam -Switch:$true'
Test-SwitchParam -Switch:$true
'Test-SwitchParam -Switch:$false'
Test-SwitchParam -Switch:$false
'$Params = @{Switch = $true}; Test-SwitchParam @Params'
$Params = @{Switch = $true}; Test-SwitchParam @Params
'$Params = @{Switch = $false}; Test-SwitchParam @Params'
$Params = @{Switch = $false}; Test-SwitchParam @Params
```

```none
Test-SwitchParam
Switch: False
Switch.IsPresent: False
  
Test-SwitchParam -Switch
Switch: True
Switch.IsPresent: True
  
Test-SwitchParam -Switch:$true
Switch: True
Switch.IsPresent: True
  
Test-SwitchParam -Switch:$false
Switch: False
Switch.IsPresent: False
  
$Params = @{Switch = $true}; Test-SwitchParam @Params
Switch: True
Switch.IsPresent: True
  
$Params = @{Switch = $false}; Test-SwitchParam @Params
Switch: False
Switch.IsPresent: False
```

> I think we should remove `NoTypeInformation` everywhere in code and leave only its declaration - the parameter is dummy and we should ignore it in code at all.

I'm torn on that. It's still an active parameter and setting it still has impact and will continue to do so. It's just hidden in favor of the new switch to make the new default behavior more clear. The internal logic would all need to be reworked. I didn't see value in switching all the logic to use `IncludeTypeInformation` when all that needs to be done is the upfront logic of detecting if both were supplied and setting `NoTypeInformation` based on the presence and setting of `-IncludeTypeInformation`. The logic would still need to be there, so why disturb the rest of the code for it?

Reply:
I see - somebody can use ` -NoTypeInformation:$false`.
However, I think we should make the code simpler and more readable. Currently  `-IncludeTypeInformation:$false` don't work. I vote for removing many private variables and using `IsPresent`.

Reply:
> Currently -IncludeTypeInformation:$false don't work

It does currently work, that's not new functionality. The decision by the committee was to maintain that functionality. This is how we preserve that functionality while adding a new parameter that is basically a reverse switch for the existing one. You can't move to `IsPresent` without breaking current functionality.

Reply:
@iSazonov I think I can make us both happy using `this.MyInvocation.BoundParameters.ContainsKey()`. I'll work on it.

Reply:
fixed

Reply:
After nice walking and good dinner I was ready to agree with you 😄